src/global_mapper/global_mapper.cc

```diff
@@ -92,6 +92,51 @@ void GlobalMapper::InsertPointCloud(const PointCloud::ConstPtr& cloud_ptr) {
   }
 }
 
+void GlobalMapper::FlattenMap() {
+  int ixyz[3] = {0};
+  double xyz[3] = {0.0};
+  double xy[2] = {0.0};
+  float occ = 0.0;
+  float occ_temp = 0.0;
+  for (int i=0; i < voxel_map_ptr_->dimensions[0]; ++i) {
+    ixyz[0] = i;
+
+    for (int j=0; j < voxel_map_ptr_->dimensions[1]; ++j) {
+      ixyz[1] = j;
+
+      // reset mean
+      occ = 0.0;
+      occ_temp = 0.0;
+
+      for (int k=0; k < voxel_map_ptr_->dimensions[2]; ++k) {
+        ixyz[2] = k;
+
+        // get coordinates from voxel_map
+        voxel_map_ptr_->tableToWorld(ixyz, xyz);
+
+        // check bounds
+        if (xyz[2] < params_.pixel_min_z_abs_ ||
+            xyz[2] > params_.pixel_max_z_abs_) {
+          continue;
+        }
+
+        // printf("z: %0.2f", xyz[2]);
+
+        // get max over z
+        occ_temp = voxel_map_ptr_->readValue(ixyz);
+        if (occ_temp > occ) {
+          occ = occ_temp;
+        }
+      }
+
+      // insert into pixel map by position
+      xy[0] = xyz[0];
+      xy[1] = xyz[1];
+      pixel_map_ptr_->writeValue(xy, occ);
+    }
+  }
+}
+
 std::unique_lock<std::mutex> GlobalMapper::CloudLock() {
   return std::unique_lock<std::mutex>(cloud_mutex_);
```

Comment:
Alright, switched all the other locks back to lock_guards given the standard on the page below.  99ad2266491c6ad7049b480c19cc826ccfd959ea.

http://www.codingstandard.com/rule/18-3-4-only-use-stdunique_lock-when-stdlock_guard-cannot-be-used/

C++ is super annoying.  It has everything from a butter knife to a battle axe, but everyone gets all out of shape if you use the battle axe to cut the cheese.  I always wish they would just make one utility that supersedes others instead of making you carefully pick from a huge array of tools to tailor to the specific situation.